lib/services.js

```diff
@@ -305,12 +308,16 @@ const services = {
                             log.getSerializedUids());
                     if (objectMD.location === null) {
                         return cb(null, res);
-                    } else if (!Array.isArray(objectMD.location)) {
-                        data.delete(objectMD.location, deleteLog);
-                        return cb(null, res);
                     }
-                    return data.batchDelete(objectMD.location, null, null,
-                    deleteLog, err => {
+
+                    const locations = Array.isArray(objectMD.location)
+                        ? objectMD.location : [objectMD.location];
+
+                    if (deferDeletion) {
+                        return cb(null, locations);
+                    }
+
+                    return data.batchDelete(locations, null, null, deleteLog, err => {
```

Comment:
code used to call `data.delete` here instead of `batchDelete` : not sure we should change that... This is probably a minor optimisation, but still...

Reply:
Done https://github.com/scality/cloudserver/pull/5195/commits/21d22ad1832687ed188f284dcafeaf6ae1c4f285#diff-a76cecfcb0b408028fb3c7e02359a47730668d6424f8b02bb9522c162c21ce47R312